examples/benchmark.py

```
@@ -0,0 +1,503 @@
+    metadata: Dict[str, Any] = field(default_factory=dict)
+
+
+class DeterministicLLM(BaseLLM):
+    """Mock LLM for deterministic benchmarking."""
+
+    def __init__(self, response_time_ms: float = 100):
+        super().__init__(model="deterministic-mock", temperature=0.0)
+        self.response_time_ms = response_time_ms
+        self.call_count = 0
+
+    def completion(self, messages: List[Dict[str, str]], **kwargs) -> str:
```

Ruff (0.14.11)

Unused method argument: `messages`

(ARG002)